pkg/controller/performanceprofile/status.go

```diff
@@ -144,3 +148,67 @@ func (r *ReconcilePerformanceProfile) getProgressingConditions(reason string, me
 		},
 	}
 }
+
+func (r *ReconcilePerformanceProfile) getMCPConditionsByProfile(profile *performancev1alpha1.PerformanceProfile) ([]conditionsv1.Condition, error) {
+
+	mcpList := &mcov1.MachineConfigPoolList{}
+
+	if err := r.client.List(context.TODO(), mcpList); err != nil {
+		klog.Errorf("Cannot list Machine config pools to match with profile %q : %v", profile.Name, err)
+		return nil, err
+	}
+
+	mcpItems := removeMCPDuplicateEntries(mcpList.Items)
+
+	performanceProfileLabels := labels.Set(profileutil.GetMachineConfigPoolSelector(profile))
+
+	reason := bytes.Buffer{}
+	reason.WriteString("Matching Machine Config Pools are in a degraded state")
```

Comment:
`reason` should be a short, predefined, machine readable string IMHO.
I see that the existing reasons (see consts on top of this file) also do not fully comply with this...
I'd suggest to add a new reason and modify existing one like this:
```
	conditionReasonValidationFailed         = "ValidationFailed"
	conditionReasonComponentsCreationFailed = "ComponentCreationFailed"
	conditionReasonMCPDegraded = "MCPDegraded"
 ```
In the message you can give the detailed dynamic info.